contracts/account/extensions/AccountERC6900.sol

```diff
@@ -0,0 +1,494 @@
+import {Bytes} from "@openzeppelin/contracts/utils/Bytes.sol";
+import {Packing} from "@openzeppelin/contracts/utils/Packing.sol";
+import {Address} from "@openzeppelin/contracts/utils/Address.sol";
+import {Calldata} from "@openzeppelin/contracts/utils/Calldata.sol";
+import {ERC165Checker} from "@openzeppelin/contracts/utils/introspection/ERC165Checker.sol";
+import {IERC6900ModularAccount, IERC6900Module, IERC6900ValidationModule, IERC6900ExecutionModule, ValidationConfig, ModuleEntity, ValidationFlags, HookConfig, ExecutionManifest, ManifestExecutionHook, ManifestExecutionFunction, Call} from "../../interfaces/IERC6900.sol";
+import {ERC6900Utils} from "../utils/ERC6900Utils.sol";
+import {ERC7739} from "../../utils/cryptography/ERC7739.sol";
+import {AccountCore} from "../AccountCore.sol";
+
+/**
+ * @dev Extension of {AccountCore} that implements support for ERC-6900 modules.
+ *
+ * To comply with the ERC-1271 support requirement, this contract implements {ERC7739} as an
+ * opinionated layer to avoid signature replayability across accounts controlled by the same key.
+ *
+ * This contract does not implement validation logic for user operations since these functionality
+ * is often delegated to self-contained validation modules. Developers must install a validator module
+ * upon initialization (or any other mechanism to enable execution from the account):
+ *
+ * ```solidity
+ * contract MyAccountERC6900is AccountERC6900, Initializable {
+ *     constructor() EIP712("MyAccount", "1") {}
+ *
+ *   function installValidation(
+ *       ValidationConfig validationConfig,
+ *       bytes4[] calldata selectors,
+ *       bytes calldata installData,
+ *       bytes[] calldata hooks
+ *   ) public initializer {
+ *     _installValidation(validationConfig, selectors, installData, hooks);
+ *   }
+ * }
+ * ```
+ *
+ */
+
+abstract contract AccountERC6900 is AccountCore, ERC7739, IERC6900ModularAccount {
+    using Bytes for *;
+    using ERC6900Utils for *;
+    using ERC6900Utils for ModuleEntity;
+    using EnumerableSet for *;
+    using Packing for bytes32;
+
+    mapping(ModuleEntity moduleEntity => Validation) private _validations;
+    mapping(bytes4 selector => Execution) private _executions;
+    mapping(bytes4 interfaceId => bool supported) private _interfaceIds;
+    mapping(bytes4 selector => address) private _fallbacks;
+
+    struct Validation {
+        EnumerableSet.Bytes32Set selectors;
+        ValidationFlags validationFlags;
+        EnumerableSet.Bytes32Set validationHooks;
+        EnumerableSet.Bytes32Set executionHooks;
+    }
+
+    struct Execution {
+        address module;
+        bool skipRuntimeValidation;
+        bool allowGlobalValidation;
+        EnumerableSet.Bytes32Set executionHooks;
+    }
+
+    /// @dev See {_fallback}.
+    fallback(bytes calldata) external payable virtual returns (bytes memory) {
+        return _fallback();
+    }
```

Comment:
you should probably pass the input.
```suggestion
    fallback(bytes calldata data) external payable virtual returns (bytes memory) {
        return _fallback(data);
    }
```

Reply:
Note this is aligned with ERC7579 impl https://github.com/OpenZeppelin/openzeppelin-community-contracts/blob/03a756ced4410403f4f11746b483c0619733c633/contracts/account/extensions/AccountERC7579.sol#L70-L72

Reply:
fair enough.